### PR TITLE
Add a pytproject.toml file

### PR DESCRIPTION
**Pull request by InnocentBug**
This hopefully defines a good build environment for plum-dispatch.

### CODE REVIEW

File: .github/workflows/quick_tests.yml [hunk 1]
```diff
@@ -23,8 +23,26 @@ jobs:
...
+                steps:
+                       - uses: actions/checkout@v2
+                       - uses: actions/setup-python@v2
+                         with:
+                             python-version: '3.x'
+                       - name: install dependecies
+                         run: |
+                           pip install jaxlib
+                       - name: install pysages
+                         run: pip install .
```
Comment (pabloferz) [inline .github/workflows/quick_tests.yml:44]: I think it needs to be run with `-e`

**Comment by InnocentBug** [comment]
Ok, so I researched it a bit.
setuptools does not fully support integration with pyproject.toml. See here for their discussion on it. https://github.com/pypa/setuptools/issues/1688

However, what they propose is to eliminate `setup.py` entirely. Which solves our issue, because if you remove the `setup.py` users are forced to install via `pip` directly.

We could specify all the config of `setup.py` in `setup.cfg` and make the build environment in `pyproject.toml`. The only thing that wouldn't work in the current scenario is the automatic version determination via `minivers`.
So if move back to manual specification of the version, I can try it out.

I am not really a fan of this, but maybe it is lesser evil.

**Comment by pabloferz** [comment]
It seems that there are efforts to deprecate installing via `setup.py` anyway. In case we do that and lead only a `pyproject.toml` and `setup.cfg`, we should be able to use this https://github.com/pypa/setuptools_scm/ as alternative to `miniver`, as it looks to be compatible with `pyproject.toml`

File: .github/workflows/quick_tests.yml (hunk 1 above)
Comment (InnocentBug) [reply .github/workflows/quick_tests.yml:28]: `-e` is for editable build.
That's a feature for developing, it has to work without it too.

**Comment by InnocentBug** [comment]
Good idea to switch to the setuptools_scm.

I implemented it and could drop the `setup.py` entirely.
A bit ironic, because this is what I originally wanted, but `pyproject.toml` seems to be the future.

**Review by pabloferz — APPROVED** [review]

**Pull request merged**